pkg/controller/framework/shared_informer.go

```diff
@@ -75,6 +77,52 @@ func NewSharedIndexInformer(lw cache.ListerWatcher, objType runtime.Object, resy
 	return sharedIndexInformer
 }
 
+// InformerSynced is a function that can be used to determine if an informer has synced.  This is useful for determining if caches have synced.
+type InformerSynced func() bool
+
+// syncedPollPeriod controls how often you look at the status of your sync funcs
+const syncedPollPeriod = 100 * time.Millisecond
+
+// WaitForCacheSync waits for caches to populate.  It returns true if it was successful, false
+// if the contoller should shutdown
+func WaitForCacheSync(stopCh <-chan struct{}, cacheSyncs ...InformerSynced) bool {
```

Comment:
> Looks like this can be implemented with pkg/util/wait.WaitFor.

I don't know that its net easier.  The `select` bit comes the wait func with some awkward reswizzling and the inner loop becomes the condition func.  You prefer the anonymous functions?